src/kes_dashboard_live_tiles_cli/LocalLoader.cpp

```diff
@@ -23,20 +44,150 @@ LocalLoader::~LocalLoader() {
 }
 
 
-double LocalLoader::getLastUpdated() {
-    return 0;
+list<shared_ptr<ITile>> LocalLoader::getTiles() {
+    list<shared_ptr<ITile>> tiles;
+
+    // Get from cache first.
+    if (!this->isCacheEmpty()) {
+        try {
+            tiles = this->getCachedTiles();
+        } catch(const CorruptedCacheException& e) {
+            cerr << e.what() << endl;
+        }
+    }
+    // If no tiles were loaded successfully, cache and return the default ones.
+    if (tiles.empty()) {
+        tiles = this->getDefaultTiles();
+        this->update(tiles);
+    }
+    return tiles;
+}
+
+
+void LocalLoader::update(const list<shared_ptr<ITile>>& tiles) {
+    // TODO: clear cache entirely
+
+    // Update cache with given tiles.
+    shared_ptr<JSON_Value> root = shared_ptr<JSON_Value>(
+        json_value_init_object(), json_value_free);
+    JSON_Object* data = json_value_get_object(root.get());
+
+    JSON_Value* tilesDataRoot = json_value_init_array();
+    JSON_Array* tilesData = json_value_get_array(tilesDataRoot);
+
+    json_object_set_number(data, "lastUpdated", getTimestamp());
+    json_object_set_value(data, "tilesData", tilesDataRoot);
+    for (auto &tile : tiles) {
+        json_array_append_value(tilesData, tile.get()->serialise());
+    }
+
+    create_directories(this->cacheDir);
+    if (json_serialize_to_file_pretty(root.get(), this->cachePath.c_str()) != JSONSuccess) {
+        cerr << "LocalLoader: update: Could not serialise data to "
+             << this->cachePath << endl;
+    }
+
+    this->cacheDataRoot = root;
+    this->setCacheData();
+}
+
+
+double LocalLoader::getLastUpdated() const {
+    if (!this->isCacheEmpty() &&
+        json_object_has_value_of_type(this->cacheData, "lastUpdated", JSONNumber)) {
+
+        return json_object_get_number(this->cacheData, "lastUpdated");
+    }
+    return -1;
```

Comment:
Would it be better to throw in this case rather than overload the return type?

Reply:
The reason why I wanted to return something and not throw is because we might choose to not set `lastUpdated` when initialising the cache with default tiles for the first time. Right now it always sets it, but the downside is that we will incur a `KES_COOLDOWN` because of it.